common/txmgr/types/tx_store.go

```diff
@@ -55,6 +56,19 @@ type TxStore[
 	FindTxesWithAttemptsAndReceiptsByIdsAndState(ctx context.Context, ids []big.Int, states []TxState, chainID *big.Int) (tx []*Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], err error)
 }
 
+// InMemoryInitializer encapsulates the methods that are used by the txmgr to initialize the in memory tx store.
+type InMemoryInitializer[
+	ADDR types.Hashable,
+	CHAIN_ID types.ID,
+	TX_HASH types.Hashable,
+	BLOCK_HASH types.Hashable,
+	R ChainReceipt[TX_HASH, BLOCK_HASH],
+	SEQ types.Sequence,
+	FEE feetypes.Fee,
+] interface {
+	AllTransactions(ctx context.Context, fromAddress ADDR, chainID CHAIN_ID) ([]Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], error)
```

Comment:
nit: Just a personal preference, I think we should rename this to have the action that we're doing like either `FetchAllTransactions` or `GetAllTransactions`

Reply:
Ah gotcha. In that case this is probably ok but think we should go with whatever is consistent across all of our method naming